File: c/pangram/pangram.c
#include "pangram.h"

#include <ctype.h>
#include <string.h>

bool is_pangram(const char *sentence) {

    // In case od NULL and empty
    if (!sentence) {
        return false;
    }

    // to lowercase
    char lower_sentence[strlen(sentence)];
    for (int i = 0; sentence[i] != '\0'; i++) {
        lower_sentence[i] = tolower(sentence[i]);
    }

    char *alphabet = "abcdefghijklmnopqrstuvwxyz";

    for (; *alphabet; alphabet++) {
        if (strchr(lower_sentence, *alphabet) == NULL) {
            return false;
        }
    }

    return true;
}

File: go/pangram/pangram.go
package pangram

import (    
    "strings"
)

func IsPangram(input string) bool {

    alphabet := "abcdefghijklmnopqrstuvwxyz"

    for _, c := range alphabet {
        if !strings.Contains(strings.ToLower(input), string(c)) {
            return false
        }
    }

    return true
}
